From: Jan-Pascal van Best <janpascal@vanbest.org>
Subject: [PATCH] debian/use-debian-java-libraries

Link to libraries under /usr/share/java instead of those
included with upstream tarball. Do not include the libraries
into the war file.

Signed-off-by: Thomas Koch <thomas.koch@ymc.ch>

---
 build.xml                           |   37 ++++++++++++++++++++++++++++++----
 common-build.xml                    |    4 ++-
 contrib/dataimporthandler/build.xml |   24 +++++++++++++++-------
 3 files changed, 51 insertions(+), 14 deletions(-)

diff --git a/build.xml b/build.xml
index 50ff2d8..7a8f380 100644
--- a/build.xml
+++ b/build.xml
@@ -22,7 +22,7 @@
   <import file="common-build.xml"/>
   
   <!-- 3rd party libraries for compilation -->
-  <property name="lib" location="lib" />
+  <property name="lib" value="/usr/share/java" />
 
   <!-- solr source files -->
   <property name="src" location="src" />
@@ -85,7 +85,7 @@
     <!-- no description, don't advertise -->
 
 
-    <available file="lib/lucene-core-${lucene_version}.jar"
+    <available file="${lib}/lucene-core-${lucene_version}.jar"
                property="lucene-version-ok-m2deploy-wont-fail" />
     <fail unless="lucene-version-ok-m2deploy-wont-fail">
       lib/lucene-core-${lucene_version}.jar Missing
@@ -116,7 +116,23 @@
   <!-- The compilation classpath -->
   <path id="compile.classpath">
     <fileset dir="${lib}">
-      <include name="*.jar" />
+      <include name="commons-codec.jar" />
+      <include name="commons-csv.jar" />
+      <include name="commons-fileupload.jar" />
+      <include name="commons-httpclient.jar" />
+      <include name="commons-io.jar" />
+      <include name="jetty.jar" />
+      <include name="jetty-util.jar" />
+      <include name="lucene-analyzers.jar" />
+      <include name="lucene-core.jar" />
+      <include name="lucene-highlighter.jar" />
+      <include name="lucene-queries.jar" />
+      <include name="lucene-snowball.jar" />
+      <include name="lucene-spellchecker.jar" />
+      <include name="servlet-api-2.5.jar" />
+      <include name="slf4j-api.jar" />
+      <include name="xml-apis-ext.jar" />
+      <include name="xml-apis.jar" />
     </fileset>
     <pathelement location="${dest}/common"/>
   </path>
@@ -138,9 +154,9 @@
     <path refid="compile.classpath" />
     <pathelement location="${dest}/solrj"/>
     <!-- jetty -->
-    <fileset dir="example/lib">
+    <!--fileset dir="example/lib">
       <include name="**/*.jar" />
-    </fileset>
+    </fileset-->
   </path>
     
   <!-- Compile the project. -->
@@ -339,6 +355,10 @@
       
     <pathelement location="${dest}/solr"/>
     <pathelement location="${dest}/solrj"/> <!-- include solrj -->
+    <fileset dir="/usr/share/java">
+      <include name="junit4.jar" />
+      <include name="easymock.jar" />
+    </fileset>
   </path>
 
   <path id="test.run.classpath">
@@ -349,6 +369,13 @@
     <path refid="compile.classpath.solrj" />
     
     <pathelement path="${java.class.path}"/>
+    <fileset dir="/usr/share/java">
+      <include name="junit4.jar" />
+      <include name="easymock.jar" />
+      <include name="jcl-over-slf4j.jar" />
+      <include name="slf4j-jdk14.jar" />
+      <include name="lucene-memory.jar" />
+    </fileset>
   </path>
 
   <!-- Compile unit tests. -->
diff --git a/common-build.xml b/common-build.xml
index c71d1b6..3e43ae6 100644
--- a/common-build.xml
+++ b/common-build.xml
@@ -62,7 +62,7 @@
        By default, this should be set to "X.Y.N-dev" where X.Y.N is
        "1 greater" then the last version released (on this branch).
     -->
-  <property name="version" value="1.4.1-dev" />
+  <property name="version" value="1.4.0" />
   
   <!-- Solr Specification Version -->
   <!--
@@ -219,7 +219,9 @@
       <subant target="@{target}" failonerror="@{failonerror}">
         <fileset dir="."
                  includes="contrib/*/build.xml"
+                 excludes="contrib/clustering/build.xml,contrib/extraction/build.xml,contrib/velocity/build.xml"
         />
+                 <!-- The above excludes have missing build dependencies in debian -->
       </subant>
     </sequential>
   </macrodef>
diff --git a/contrib/dataimporthandler/build.xml b/contrib/dataimporthandler/build.xml
index b2832a7..6ed3cca 100644
--- a/contrib/dataimporthandler/build.xml
+++ b/contrib/dataimporthandler/build.xml
@@ -36,16 +36,24 @@
   <path id="common.classpath">
   	<pathelement location="${solr-path}/build/solr" />
   	<pathelement location="${solr-path}/build/solrj" />
-  	<fileset dir="${solr-path}/lib" includes="*.jar"/>
+    <fileset dir="/usr/share/java">
+      <include name="commons-io.jar" />
+      <include name="lucene-core.jar" />
+      <include name="slf4j-api.jar" />
+      <include name="xml-apis.jar" />
+      <include name="xml-apis-ext.jar" />
+    </fileset>
   </path>
 
   <path id="extras.classpath">
   	<pathelement location="${solr-path}/build/solr" />
   	<pathelement location="${solr-path}/build/solrj" />
   	<pathelement location="target/classes" />
-  	<fileset dir="${solr-path}/lib" includes="*.jar"/>
-  	<fileset dir="lib/" includes="*.jar"/>
-  	<fileset dir="${tikalibs-path}" includes="*.jar"/>
+  	<!--fileset dir="${tikalibs-path}" includes="*.jar"/-->
+    <fileset dir="/usr/share/java">
+      <include name="slf4j-api.jar" />
+      <include name="gnumail.jar" />
+    </fileset>
   </path>
 	
   <path id="test.classpath">
@@ -101,11 +109,11 @@
     </solr-javac>
   </target>
 	
-  <target name="build" depends="compile,compileExtras">
+  <target name="build" depends="compile">
     <solr-jar destfile="target/${fullnamever}.jar" basedir="target/classes"
               manifest="${common.dir}/${dest}/META-INF/MANIFEST.MF" />
-    <solr-jar destfile="target/apache-${ant.project.name}-extras-${version}.jar" basedir="target/extras/classes"
-              manifest="${common.dir}/${dest}/META-INF/MANIFEST.MF" />
+    <!--solr-jar destfile="target/apache-${ant.project.name}-extras-${version}.jar" basedir="target/extras/classes"
+              manifest="${common.dir}/${dest}/META-INF/MANIFEST.MF" /-->
   </target>
 	
   <target name="compileTests" depends="compile">
@@ -177,7 +185,7 @@
   	<mkdir dir="../../build/web/WEB-INF/lib"/>
   	<copy file="target/${fullnamever}.jar" todir="${solr-path}/build/web/WEB-INF/lib"></copy>
   	<copy file="target/${fullnamever}.jar" todir="${solr-path}/dist"></copy>
-  	<copy file="target/apache-${ant.project.name}-extras-${version}.jar" todir="${solr-path}/dist"></copy>
+  	<!--copy file="target/apache-${ant.project.name}-extras-${version}.jar" todir="${solr-path}/dist"></copy-->
   </target>
 	
   <target name="javadoc">
-- 
tg: (0ec0e86..) debian/use-debian-java-libraries (depends on: upstream)
